### PR TITLE
fix(TBD-12765): Move the "Use Knox" checkbox to "Connection" section

### PR DESCRIPTION
**Pull request by sanulich**

### CODE REVIEW

**Review by AlixMetivier — DISMISSED** [review]

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 1]
```diff
@@ -1235,6 +1249,27 @@ public void modifyText(ModifyEvent e) {
...
+    private void reloadForm() {
+        ((HadoopClusterForm) this.getParent()).switchToInfoForm();
+    }
+
+    private void updateKnoxPart() {
+        if(HCVersionUtil.isExecutedThroughKnox(getConnection())) {
+            hideControl(useKnoxButton, false);
+        } else {
+            hideControl(useKnoxButton, true);
+        }
```
Comment (samarie) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:1272]: you could simplify here : 
```
 hideControl(useKnoxButton, !HCVersionUtil.isExecutedThroughKnox(getConnection()) );
```

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 2]
```diff
@@ -1235,6 +1249,27 @@ public void modifyText(ModifyEvent e) {
             }
         });
 
+        useKnoxButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+
+                String selection = String.valueOf(useKnoxButton.getSelection());
+                getConnection().getParameters().put(ConnParameterKeys.CONN_PARA_KEY_USE_KNOX, selection);
```
Comment (samarie) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:1257]: Questions : 
 - could   getConnection()  be null ? ( if yes you could  have NPE ) 
 -  could  getParameters() be null ?  ( if yes you could have NPE )

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 3]
```diff
@@ -316,6 +318,11 @@ public void init() {
 
         onUseCustomConfBtnSelected(null);
         onOverrideHadoopConfBtnSelected(null);
+
+        //knox
+        String useKnoxStr = getConnection().getParameters().get(ConnParameterKeys.CONN_PARA_KEY_USE_KNOX);
```
Comment (samarie) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:323]: Questions :

    could getConnection() be null ? ( if yes you could have NPE )
    could getParameters() be null ? ( if yes you could have NPE )

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java (hunk 1 above)
Comment (sanulich) [reply main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java]: done

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java (hunk 3 above)
Comment (sanulich) [reply main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:323]: I think that getConnection() couldn't be null

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java (hunk 2 above)
Comment (sanulich) [reply main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:1257]: I think that getConnection() couldn't be null

**Pull request merged**